internal/omap/orderedmap.go

```diff
@@ -61,16 +64,31 @@ func (om *OrderedMap[K, V]) Set(key K, value V) {
 	if om.m == nil {
 		om.m = make(map[K]V)
 	}
+	if om.mutex == nil {
```

Comment:
Doesn't look right, there is a race condition here

Reply:
Yes, we should likely instantiate this mutex on the `New` function.